scripts/reliability_report.py

```diff
@@ -0,0 +1,272 @@
+            cast(OrderedDict, meta)["since"] = cast(
+                datetime, cast(OrderedDict, meta).get("since", datetime.now())
+            ).strftime("%Y-%m-%d %H:%M:%S.%f %Z")
+        except Exception as ex:
+            self.log.error(ex)
+        return json.dumps(meta, indent=2)
+
+    async def as_md(self):
+        """Return as a MarkDown formatted document"""
+        env = jinja2.Environment(
+            loader=jinja2.PackageLoader("reliability_report"),
+            autoescape=jinja2.select_autoescape(),
+        )
+        template = env.get_template("reliable_report_template.md")
+        data = await self.collect()
+        return template.render(
+            date=datetime.now().strftime(format="%Y-%m-%d %H:%M:%S %Z"),
+            meta=data.get("meta"),
+        )
+
+
+def config(env_args: os._Environ = os.environ) -> argparse.Namespace:
+    """Read the configuration from the args and environment."""
+    parser = argparse.ArgumentParser(
+        description="Manage Autopush Reliability Tracking Redis data."
+    )
+    parser.add_argument("-c", "--config", help="configuration_file", action="append")
+    parser.add_argument(
+        "--db_dsn",
+        "-b",
+        help="Database DSN connection string",
+        default=env_args.get(
+            "AUTOEND_DB_DSN",
+            env_args.get("AUTOCONNECT_DB_DSN", "grpc://localhost:8086"),
+        ),
+    )
+    parser.add_argument(
+        "--db_settings",
+        "-s",
+        help="Database settings",
+        default=env_args.get(
+            "AUTOEND_DB_SETTINGS",
+            env_args.get(
+                "AUTOCONNECT_DB_SETTINGS",
+                '{"message_family":"message","router_family":"router", "table_name":"projects/test/instances/test/tables/autopush"}',
+            ),
+        ),
+    )
+    parser.add_argument(
+        "--track_family",
+        help="Name of Bigtable reliability logging family",
+        default=env_args.get("AUTOTRACK_FAMILY", RELIABILITY_FAMILY),
+    )
+    parser.add_argument(
+        "--output",
+        help="Output format: md (MarkDown text), json (Formatted JSON), raw (python dump)",
+        default="md",
+        choices=["md", "json", "raw"],
+    )
+    args = parser.parse_args()
+
+    # if we have a config file, read from that and then reload.
+    if args.config is not None:
+        for filename in args.config:
+            with open(filename, "r") as f:
+                args = parser.set_defaults(**toml.load(f))
+    args = parser.parse_args()
+    # fixup the bigtable settings so that they're easier for this script to deal with.
+    if args.db_settings is not None:
+        bt_settings = json.loads(args.db_settings)
+        parts = bt_settings.get("table_name").split("/")
+        for i in range(0, len(parts), 2):
+            # remember: the `tablename` dsn uses plurals for
+            # `projects`, `instances`, & `tables`
+            bt_settings[parts[i].rstrip("s")] = parts[i + 1]
```

Comment:
You're saying "projects/etc/instances/etcetc" isn't compatible with this client? Isn't that the standard? e.g. `_INSTANCE_NAME_RE` [here](https://gcloud.readthedocs.io/en/latest/_modules/google/cloud/bigtable/instance.html)

Reply:
It is. 
What we're doing here is converting the plural names to singular keys e.g. `projects/foo/instances/bar/...` become `"project"=>"foo", "instance"=>"bar"` we use this later in `BigtableScanner.collect()` 
Granted, we could have just as easily used the plural and instead gone with 
```python
table = self.bigtable.get_table(
    self.settings.bigtable.get("instances"),
    self.settings.bigtable.get("tables"),
)
```
but that parses a bit strange, since these are singular values.